src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -79,20 +77,20 @@ internal sealed class VariableState
         /// Method signature used for return type or parameter types. Distinct from _member
         /// signature when _member is a lambda and type is inferred from MethodTypeInferrer.
         /// </summary>
-        private readonly MethodSymbol _methodSignatureOpt;
+        private readonly MethodSymbol? _methodSignatureOpt;
 
         /// <summary>
         /// Types from return expressions. Used when inferring lambda return type in MethodTypeInferrer.
         /// </summary>
-        private readonly ArrayBuilder<(RefKind, TypeSymbolWithAnnotations)> _returnTypes;
+        private readonly ArrayBuilder<(RefKind, TypeSymbolWithAnnotations)>? _returnTypes;
```

Comment:
Slightly surprising, should this be named `_returnTypesOpt`?

Reply:
Renamed in [https://github.com/dotnet/roslyn/pull/31771](https://github.com/dotnet/roslyn/pull/31771) rather than here since it will be a while before this branch can be merged into master.

---
In reply to: [241890092](https://github.com/dotnet/roslyn/pull/31719#discussion_r241890092) [](ancestors = 241890092)